include/MemoryManager.h

```diff
@@ -103,6 +103,25 @@ class EXPORT MemoryManager
 	static QMutex s_pointerMutex;
 };
 
+template<typename T>
+struct MmAllocator
+{
+	typedef T value_type;
+	template< class U > struct rebind { typedef MmAllocator<U> other; };
+
+	T* allocate( std::size_t n )
+	{
+		return reinterpret_cast<T*>( MemoryManager::alloc( sizeof(T) * n ) );
+	}
+
+	void deallocate( T* p, std::size_t )
+	{
+		MemoryManager::free( p );
+	}
+
+	typedef std::vector< T, MmAllocator<T> > vector;
+};
+
```

Comment:
These parts aren't used. Are there some reasons you leave them in our codebase?
Some more inconsistencies: you mixed `< foo >` and `<foo>`, it makes the code looks inconsistent.

Reply:
> Are there some reasons you leave them in our codebase?

I plan on using them at a later point in time. Using an [Allocator](http://en.cppreference.com/w/cpp/concept/Allocator) class is the standard C++ way of having custom allocators, and should be preferred over using the macros `MM_ALLOC` and `MM_FREE`.

> you mixed < foo > and <foo>, it makes the code look inconsistent.

This has already been discussed by @Umcaruje and me in the comments here. C++03 doesn't allow `>>` for nested template arguments (it always interprets it as the stream operator `>>`). We can't use `>>` here, because the headers is included by plugins that are compiled without C++11 support. See d1aa39bfd194b0645aeadb647808c7ed15ce8f7e.

Reply:
> if you are going to use this later, I think vector is not very good name

Well it's a vector, so why not call it `vector`? What would you call it?

Reply:
You're right, but you used both `template<typename T>` and `template< class U >` above. Is that intentional?

Reply:
Ah, no that's not intentional.